pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/AnswerDao.java

```diff
@@ -263,6 +247,13 @@ default int deleteAllByInstanceIds(Set<Long> instanceIds) {
     @UseRowReducer(AnswerWithValueReducer.class)
     Optional<Answer> findAnswerByGuid(@Bind("guid") String answerGuid);
 
+    @UseStringTemplateSqlLocator
+    @SqlQuery("queryAnswerForInstanceIdAndQuestionStableId")
+    @UseRowReducer(AnswerWithValueReducer.class)
+    Optional<Answer> findAnswerForInstanceIdAndQuestionStableId(
+            @Bind("instanceId") long instanceId,
+            @Bind("questionStableId") String questionStableId);
+
```

Comment:
This query is somewhat unique, particular the composite part of it. It also looks really useful. Surprised we did not have something like this before.
You might want to consider creating a unit tests for this one method that checks composite. You already touching in your PEX test so that is good, but in isolation would be better and having a case that tries out composite would give some additional assurances.

Reply:
Good idea. Added some tests